vault_cli/cli.py

```diff
@@ -451,7 +452,11 @@ def mv(
 
 
 @cli.command()
-@click.argument("template", type=click.File("r"), required=True)
+@click.argument(
+    "template",
+    type=click.Path(exists=True, allow_dash=True, file_okay=True),
+    required=True,
+)
```

Comment:
I'm a bit surprised that it doesn't work keeping click.File(). I would have guessed that when a filepath is passed, the resulting object has a `.name` attribute that point to the filename, and when `-` is passed, the resulting object has no `.name` attribute.

Reply:
I didn't really try to be honest. if it even works, that would still be an undocumented feature: https://click.palletsprojects.com/en/7.x/arguments/#file-args 

Plus, the bonus is:

>  Not only will it return either bytes or Unicode depending on what makes more sense, but it will also be able to do some basic checks for you such as existence checks.

Reply:
Well so does `click.File` :)

I'm reasonably comfortable following your choice and using click.Path then. Also, see https://github.com/pallets/click/issues/188

Reply:
Thanks!